jc/parsers/lsattr.py

```diff
@@ -0,0 +1,157 @@
+"""jc - JSON Convert `lsattr` command output parser
+
+Usage (cli):
+
+    $ lsattr | jc --lsattr
+
+or
+
+    $ jc lsattr
+
+Usage (module):
+
+    import jc
+    result = jc.parse('lsattr', lsattr_command_output)
+
+Schema:
+
+Information from https://github.com/mirror/busybox/blob/2d4a3d9e6c1493a9520b907e07a41aca90cdfd94/e2fsprogs/e2fs_lib.c#L40
+used to define field names
+
+    [
+      {
+        "file":                           string,
+        "Compressed_File":                Optional[boolean],
+        "Compressed_Dirty_File":          Optional[boolean],
+        "Compression_Raw_Access":         Optional[boolean],
+        "Secure_Deletion":                Optional[boolean],
+        "Undelete":                       Optional[boolean],
+        "Synchronous_Updates":            Optional[boolean],
+        "Synchronous_Directory_Updates":  Optional[boolean],
+        "Immutable":                      Optional[boolean],
+        "Append_Only":                    Optional[boolean],
+        "No_Dump":                        Optional[boolean],
+        "No_Atime":                       Optional[boolean],
+        "Compression_Requested":          Optional[boolean],
+        "Encrypted":                      Optional[boolean],
+        "Journaled_Data":                 Optional[boolean],
+        "Indexed_directory":              Optional[boolean],
+        "No_Tailmerging":                 Optional[boolean],
+        "Top_of_Directory_Hierarchies":   Optional[boolean],
+        "Extents":                        Optional[boolean],
+        "No_COW":                         Optional[boolean],
+        "Casefold":                       Optional[boolean],
+        "Inline_Data":                    Optional[boolean],
+        "Project_Hierarchy":              Optional[boolean],
+        "Verity":                         Optional[boolean],
+      }
+    ]
+
+Examples:
+
+      $ sudo lsattr /etc/passwd | jc --lsattr
+      [
+        {
+            "file": "/etc/passwd",
+            "Extents": true
+        }
+      ]
+"""
+
+import jc.utils
+
+
+class info():
+    """Provides parser metadata (version, author, etc.)"""
+    version = '1.0'
+    description = '`lsattr` command parser'
+    author = 'Kelly Brazil'
+    author_email = 'kellyjonbrazil@gmail.com'
+    compatible = ['linux', 'darwin', 'cygwin', 'aix', 'freebsd']
+    magic_commands = ['lsattr']
+    tags = ['command']
+
+
+__version__ = info.version
+
+
+ERROR_PREFIX = "lsattr:"
+
+# https://github.com/mirror/busybox/blob/2d4a3d9e6c1493a9520b907e07a41aca90cdfd94/e2fsprogs/e2fs_lib.c#L40
+# https://github.com/landley/toybox/blob/f1682dc79fd75f64042b5438918fe5a507977e1c/toys/other/lsattr.c#L97
+ATTRIBUTES = {
+    "B": "Compressed_File",
+    "Z": "Compressed_Dirty_File",
+    "X": "Compression_Raw_Access",
+    "s": "Secure_Deletion",

```

Comment:
Could we lower-case the dict values here? Makes it nicer to use via command-line so the JSON key-names will be all lower with underscores.